src/sentry/integrations/msteams/webhook.py

```diff
@@ -302,7 +230,7 @@ def _get_team_installation_request_data(cls, data: dict[str, Any]) -> dict:
         }
         return params
 
-    def handle_installation_update_event(self, request: HttpRequest) -> HttpResponse:
+    def _handle_installation_update_event(self, request: Request) -> Response:
```

Comment:
Was `HttpRequest/Response` not accurate before? I'm mainly curious why this change was necessary.

Reply:
Basically yes. Anywhere you see `request.data`, you're relying on it being a `rest_framework.requests.Request`. But the inheritance structure of our `Endpoint` class is kind of messed up per [this comment thread](https://github.com/getsentry/sentry/pull/77586#discussion_r1761799185).